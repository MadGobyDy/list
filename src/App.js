import './App.scss';
import List from "./list";
import ListForm from "./listForm"
import { db } from "./firebase";
import {useEffect} from "react";
import {doc, onSnapshot} from "firebase/firestore";
import {useDispatch} from "react-redux";
import {initial} from './redux';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    onSnapshot(doc(db, "fruits", "fruits"), (doc) => {
      dispatch(initial(doc.data().items));
    });
  }, [dispatch])

  return (
    <div className="app">
      <ListForm />
      <List />
    </div>
  );
}

export default App;
