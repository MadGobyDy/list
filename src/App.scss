@import "~bootstrap/scss/bootstrap";

.app {
  padding: 40px 60px;
}