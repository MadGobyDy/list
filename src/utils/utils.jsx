import React from 'react'
import { render } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
// As a basic setup, import your same slice reducers
import listReducer from '../store/listSlice'

export function renderWithProviders(
  ui,
  {
    preloadedState = {list: {items: []}},
    store = configureStore({ reducer: { list: listReducer }, preloadedState }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}