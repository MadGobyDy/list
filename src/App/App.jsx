import React, { useEffect } from 'react';
import './App.scss';
import { doc, onSnapshot } from 'firebase/firestore';
import { useDispatch } from 'react-redux';
import { List } from '../components/List/List';
import ListForm from '../components/ListForm/ListForm';
import { db } from '../firebase/firebase';
import { initial } from '../store/listSlice';

function App() {
    const dispatch = useDispatch();

    useEffect(() => {
        onSnapshot(doc(db, 'fruits', 'fruits'), (docInst) => {
            dispatch(initial(docInst.data().items));
        });
    }, [dispatch]);

    return (
        <div className="app" data-testid="app">
            <ListForm />
            <List />
        </div>
    );
}
export { App };
