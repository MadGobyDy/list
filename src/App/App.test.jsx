import { renderWithProviders } from '../utils/utils'
import { App } from "./App";
import { screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';

describe('App tests', () => {

    test('App to match snapshot', () => {
        renderWithProviders(<App />);

        const app = screen.getByTestId('app')

        expect(app).toMatchSnapshot();
    });

    test('Input in the doc', () => {
        renderWithProviders(<App />);

        const input = screen.getByRole('textbox');
        expect(input).toBeInTheDocument();
    });

    test('list in the doc', () => {
        renderWithProviders(<App />);

        const list = screen.getByTestId('list');
        expect(list).toBeInTheDocument();
    });

    test('The string sent through the form gets into the list', () => {
        const { store } = renderWithProviders(<App />);

        expect(store.getState().list.items).toEqual([]);

        const emptyList = screen.getByTestId('emptyList');
        expect(emptyList).toBeInTheDocument();

        const input = screen.getByRole('textbox');
        userEvent.type(input, 'apple');
        const submitButton = screen.getByTestId('inputButton');
        userEvent.click(submitButton);

        expect(store.getState().list.items).toEqual(['apple']);

        const apple = screen.getByText('apple');
        expect(apple).toBeInTheDocument();
    });
});