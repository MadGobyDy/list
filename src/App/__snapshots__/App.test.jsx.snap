// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests App to match snapshot 1`] = `
<div
  class="app"
  data-testid="app"
>
  <div
    class="listForm"
  >
    <div
      class="input-group"
    >
      <span
        class="listForm__text"
      >
        Добавить фрукт в список: 
      </span>
      <input
        class="form-control"
        placeholder="Напишите название фрукта"
      />
      <button
        class="btn btn-primary"
        data-testid="inputButton"
        type="button"
      >
        Добавить
      </button>
    </div>
  </div>
  <div
    class="list list-group"
    data-testid="list"
  >
    <span
      class="list__empty"
      data-testid="emptyList"
    >
      Список фруктов пуст
    </span>
  </div>
</div>
`;
